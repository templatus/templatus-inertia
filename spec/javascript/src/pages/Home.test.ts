const mockSendClick = jest.fn();
const mockUnsubscribe = jest.fn();
const mockGetClicks = jest.fn();
const mockSubscribe = jest.fn();

jest.mock('@/stores/click', () => ({
  __esModule: true,
  useClickStore: () => ({
    loaded: true,
    total: 42,
    items: [
      {
        created_at: '2021-05-23T09:27:21.497Z',
        ip: '1.2.3.4',
        user_agent: 'Jest',
      },
    ],
    getClicks: mockGetClicks,
    subscribe: mockSubscribe,
    unsubscribe: mockUnsubscribe,
    sendClick: mockSendClick,
  }),
}));

import { mount } from '@vue/test-utils';
import Home from '@/pages/HomePage.vue';

describe('Home', () => {
  const wrapper = mount(Home, {
    props: {
      name: 'World',
    },
  });

  test('load clicks and subscribes', () => {
    expect(mockSubscribe).toHaveBeenCalled();
    expect(mockGetClicks).toHaveBeenCalled();
  });

  test('renders name', () => {
    const title = wrapper.find('h1')?.element?.textContent?.trim();
    expect(title).toEqual('Hello, World!');
  });

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('executes sendClick', () => {
    const button = wrapper.find('button');
    button.trigger('click');

    expect(mockSendClick).toHaveBeenCalled();
  });

  test('unsubscribes on unmount', () => {
    wrapper.unmount();
    expect(mockUnsubscribe).toHaveBeenCalled();
  });
});
