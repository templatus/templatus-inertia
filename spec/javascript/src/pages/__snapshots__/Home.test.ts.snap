// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div class="lg:mr-64">
  <h1 class="break-words text-4xl font-extrabold tracking-tight text-rails-dark sm:text-5xl lg:text-6xl">Hello, World!</h1>
  <p class="mt-6 max-w-4xl text-base lg:text-xl"> Templatus is an opinionated template for web applications with <b>Ruby on Rails</b> and <b>Vue.js 3</b></p>
</div>
<div class="flex flex-col mt-5 w-full lg:absolute lg:-top-5 lg:-right-10 lg:mt-0 lg:w-56 lg:rounded lg:bg-gradient-to-br lg:from-slate-100 lg:to-white lg:p-3 lg:shadow">
  <div id="counter" class="text-center">
    <transition-stub>
      <div class="text-5xl font-bold">42</div>
    </transition-stub>
    <div class="py-1 uppercase">Clicks</div>
  </div><button type="button" class="mt-3 w-full rounded bg-rose py-2 font-medium text-black hover:bg-rails-light hover:text-white active:outline-none active:ring-2 active:ring-rails-light active:ring-offset-1"> Click me! </button>
  <p class="mt-3 hidden text-center text-xs text-gray-600 sm:block"> Tip: Open two browser windows in parallel and see what happens. </p>
</div>
<h2 class="mt-20 text-center text-xl font-medium uppercase tracking-wide text-rails-dark sm:text-left"> Latest clicks </h2>
<transition-group-stub>
  <li class="py-4">
    <p><span class="rounded bg-gray-900 py-1 px-2 font-mono text-rose">1.2.3.4</span><span class="ml-2 px-1 font-medium text-gray-900">Sun, 23 May 2021 09:27:21 GMT</span></p><code class="mt-4 block font-extralight text-gray-500">Jest</code>
  </li>
</transition-group-stub>
`;
